### PR TITLE
qemu: apply `hvf: arm: disable SME which is not properly handled by QEMU`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
From https://github.com/utmapp/UTM/blob/v4.6.3/patches/qemu-9.1.2-utm.patch#L714-L741

Needed by macOS 15.2 (or later) on Apple M4 (or later):
- https://gitlab.com/qemu-project/qemu/-/issues/2665
- https://gitlab.com/qemu-project/qemu/-/issues/2721

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/q/qemu.rb [hunk 1]
```diff
@@ -62,6 +63,13 @@ class Qemu < Formula
     depends_on "systemd"
   end
 
+  # hvf: arm: disable SME which is not properly handled by QEMU
+  # From https://github.com/utmapp/UTM/blob/v4.6.3/patches/qemu-9.1.2-utm.patch#L714-L41
+  # Needed by macOS 15.2 (or later) on Apple M4 (or later)
+  # https://gitlab.com/qemu-project/qemu/-/issues/2665
+  # https://gitlab.com/qemu-project/qemu/-/issues/2721
+  patch :DATA
```
Comment (SMillerDev) [inline Formula/q/qemu.rb:71]: Can we not use the upstream patch?

Comment (AkihiroSuda) [reply Formula/q/qemu.rb:71]: Which patch?

Comment (AkihiroSuda) [reply Formula/q/qemu.rb:71]: I don't think there is an upstream patch.
The issue still happens with `brew install --HEAD` (HEAD-3e9793a):
```console
$ qemu-system-aarch64 -M virt -cpu host -accel hvf
Unexpected error in object_property_find_err() at ../qom/object.c:1345:
qemu-system-aarch64: Property 'host-arm-cpu.sme' not found       
Abort trap: 6 
```

Comment (daeho-ro) [reply Formula/q/qemu.rb:71]: Yeap, I haven't found any patch file and so we cannot use a patch url.

**Review by daeho-ro — APPROVED** [review]

**Review by SMillerDev — COMMENTED** [review]
It worries me that nobody upstream has even responded to these issues yet, and the patch we use is one marked "DON NOT MERGE" from a completely different project.
If we're going to merge a patch here I'd feel much more comfortable if it was upstreamed in any way, especially considering the importance of the project.

**Pull request closed without merging**